pages/settings.tsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import * as Styled from "../styles/settingStyles"
+import { Button } from '../src/components/Button';
+import { Footer } from '../src/components/footerNavBar';
+import { useAccountDetails } from '../src/hooks/useAccountDetail';
+
+export default function settings() {
+    const hook = useAccountDetails();
+    return (
+        <Styled.SettingsPage>
+            <Styled.Header>
+                <Styled.WelcomeNote>
+                    Welcome Hashim!
+                </Styled.WelcomeNote>
+                <Button theme='TextBoxHome' height="40px" width="fit-content" buttonText="SIGN OUT" buttonIcon="" onClick={() => console.log('clicked')} />
+            </Styled.Header>
+            <Styled.AccountDetailsText> Account Details</Styled.AccountDetailsText>
+            <Styled.InputSection>
+                <Styled.FeildContainer>
+                    <Styled.InputLabel> First Name </Styled.InputLabel>
+                    <Styled.EditableInput type='text' value={hook.firstName} onChange={(e) => hook.setFirstName(e.target.value)} />
```

Comment:
make separate on change functions for each input in the hook file and export them from there. You never know if you will need to change something in the hook as well. Rmbr all computations are to be made in the hook file.